xapi/storage/libs/libcow/volume.py

```diff
@@ -273,7 +273,10 @@ def stat(dbg, sr, key, cb):
                 _vdi_sanitize(vdi, opq, db, cb)
                 custom_keys = db.get_vdi_custom_keys(vdi.uuid)
 
-            psize = cb.volumeGetPhysSize(opq, str(vdi.volume.id))
+            psize = 0
+            if cb.volumeExists(opq, str(vdi.volume.id)):
+                psize = cb.volumeGetPhysSize(opq, str(vdi.volume.id))
+
```

Comment:
IMHO, we should avoid this modification, this code is called in many drivers.
It's normal to raise an exception if the size cannot be fetched.

Reply:
I need to double-check this change of behavior. When triggering `sr.scan()`, I observed that if the volume does not exist `sr.ls()` crashes without calling `sr.stat()`. I had somehow to prevent that crash otherwise `sr.scan()` keeps calling `sr.ls()` and crashing.

Reply:
I moved stat to the driver.